### PR TITLE
Fix bug: divide by zero in embedding_lookup_sparse

### LINKED ISSUE #14851: Got NAN when calling embedding_lookup_sparse with weights and "mean" combiner
### System information
- **OS Platform and Distribution**: Win7 64bit
- **TensorFlow installed from**: anaconda binary
- **TensorFlow version**: 1.2.1
- **Python version**: 3.6

### Describe the problem
I always got nan when I call embedding_lookup_sparse with weights and "mean" combiner.
Code pieces are listed below:

### Source code / logs
from __future__ import absolute_import
from __future__ import print_function
import os
import numpy as np
import tensorflow as tf
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'

a = np.arange(192).reshape(24, 8)
print(a)

a = tf.Variable(a, dtype=tf.float32)
ids = tf.SparseTensor(
    indices=[[1, 0], [2, 0], [2, 1], [2, 2], [3, 0], [9, 1]],
    values=[10, 1, 2, 3, 4, 5],
    dense_shape=[1, 1])
weights = tf.SparseTensor(
    indices=[[1, 0], [2, 0], [2, 1], [2, 2], [3, 0], [9, 1]],
    values=[1, 0.2, 0.4, 0.4, 1, 1],
    dense_shape=[1, 1])
b = tf.nn.embedding_lookup_sparse(
    a, ids, weights, partition_strategy='mod', combiner='sum')
c = tf.pad(b, [[0, 16 - tf.shape(b)[0]], [0, 0]], mode='CONSTANT')

sess = tf.Session()
sess.run(tf.global_variables_initializer())
[value] = sess.run([c])
print(value)

### PR DESCRIPTION
**Pull request by facaiy**
Fix #14851

### How to test

+ [x] add test case
+ [ ] pass all tests.

### CODE REVIEW

**Comment by facaiy** [comment]
Hi, @sb2nov . All python tests are passed on my working machine (Linux, CPU, Python2). I cannot check those failed logs because now Details links to a web which seems private for google.

**Comment by drpngx** [comment]
There was some infra error, restarting.

**Comment by facaiy** [comment]
Thanks, @drpngx ! Ubuntu CC failed with "infrastructure error", I cannot figure out what it means?

**Comment by drpngx** [comment]
Yeah, we are having issues with the infra.

**Comment by facaiy** [comment]
Hi, @drpngx . I merged the latest master branch, which might be helpful to pass all tests.

**Comment by drpngx** [comment]
Jenkins, test this please

**Comment by drpngx** [comment]
Could you check the test errors?

**Comment by facaiy** [comment]
Thank you, @drpngx . Could you paste the failure log? It looks like a timeout case.
```bash
//bazel_pip/tensorflow/contrib/timeseries/python/timeseries:input_pipeline_test FAILED in 14.7s
```

**Comment by drpngx** [comment]
It's more serious than than. Could you run again with asan?

```
INFO: From Testing //bazel_pip/tensorflow/contrib/timeseries/python/timeseries:input_pipeline_test:
==================== Test output for //bazel_pip/tensorflow/contrib/timeseries/python/timeseries:input_pipeline_test:
2017-12-25 18:40:07.339780: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX
.......................
----------------------------------------------------------------------
Ran 23 tests in 10.055s

OK
Segmentation fault (core dumped)
```

**Comment by facaiy** [comment]
@drpngx Strange that the same test pass on my machine: 
+ Python 3.6
+ CentOS Linux release 7.3.1611 (Core)
+ Linux hxxx.xxx.xxx.xxx.com 3.10.0-514.10.2.el7.x86_64 #1 SMP Fri Mar 3 00:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux


```bash
cc1plus: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
Target //tensorflow/contrib/timeseries/python/timeseries:input_pipeline_test up-to-date:
  bazel-bin/tensorflow/contrib/timeseries/python/timeseries/input_pipeline_test
INFO: Elapsed time: 690.946s, Critical Path: 156.82s
//tensorflow/contrib/timeseries/python/timeseries:input_pipeline_test    PASSED in 12.6s

Executed 1 out of 1 test: 1 test passes.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
```

**Comment by drpngx** [comment]
Did you try with address and memory sanitizers?

**Comment by facaiy** [comment]
I don't know what is address and memory sanitizers. And the test is executed with default configuration:

```bash
(py36) [facai@h1077922 tensorflow]$ cat .tf_configure.bazelrc
build --action_env PYTHON_BIN_PATH="/usr/home/facai/workshop/anaconda2/envs/py36/bin/python"
build --action_env PYTHON_LIB_PATH="/usr/home/facai/workshop/anaconda2/envs/py36/lib/python3.6/site-packages"
build --force_python=py3
build --host_force_python=py3
build --python_path="/usr/home/facai/workshop/anaconda2/envs/py36/bin/python"
build --define with_jemalloc=true
build --define with_gcp_support=true
build --define with_hdfs_support=true
build --define with_s3_support=true
build:xla --define with_xla_support=true
build:gdr --define with_gdr_support=true
build:verbs --define with_verbs_support=true
build --action_env TF_NEED_OPENCL_SYCL="0"
build --action_env TF_NEED_CUDA="0"
build --define grpc_no_ares=true
build:opt --copt=-march=native
build:opt --host_copt=-march=native
build:opt --define with_default_optimizations=true
build --copt=-DGEMMLOWP_ALLOW_SLOW_SCALAR_FALLBACK
build --host_copt=-DGEMMLOWP_ALLOW_SLOW_SCALAR_FALLBACK
build:mkl --define using_mkl=true
build:mkl -c opt
build:monolithic --define framework_shared_object=false
build --define framework_shared_object=true
build:android --crosstool_top=//external:android/crosstool
build:android --host_crosstool_top=@bazel_tools//tools/cpp:toolchain
build:android_arm --config=android
build:android_arm --cpu=armeabi-v7a
build:android_arm64 --config=android
build:android_arm64 --cpu=arm64-v8a
```

**Comment by drpngx** [comment]
You should be able to build with `bazel --config=asan -c dbg` instead of your regular command.

**Comment by drpngx** [comment]
@facaiy what do you mean "pend" the discussion?

**Comment by rmlarsen** [comment]
@drpngx I think he meant "pending the discussion of safe_div op #15706", i.e. wait for the discussion there to be resolved.

**Comment by drpngx** [comment]
Ahh! Thanks for the clarification :)

@facaiy please ping when ready.

**Comment by facaiy** [comment]
@drpngx OK, I'll let you know when ready. 
Thank you, @rmlarsen .

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @drpngx: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 59 days that this pull-request has stalled. Please create a new pull-request with the requested changes.
